README.md

```diff
@@ -36,75 +37,190 @@ pod 'nodes-ios/Rye'
 
+```swift
+import Rye
+...
 let ryeConfiguration: RyeConfiguration = [
     Rye.Configuration.Key.text: "Error message for the user",
     Rye.Configuration.Key.backgroundColor: UIColor.red.withAlphaComponent(0.4),
     Rye.Configuration.Key.animationType: Rye.AnimationType.fadeInOut]
 ]
 
-let rye = RyeViewController.init(alertType: .toast,
-                                viewType: .standard(configuration: ryeConfiguration),
-                                at: .bottom(inset: 16),
-                                timeAlive: 2)
+let rye = RyeViewController.init(viewType: .standard(configuration: ryeConfiguration),
+                                 at: .bottom(inset: 16))
 rye.show()
 
 ```
 
 ### Display Rye with a Custom `UIView`
 
-```swift
+For even more control you can create your own subclass of `UIView` and use `.custom` for the `viewType` parameter
 
-let customRyeView = RyeView()
+```swift
+import Rye
+...
 
-let rye = RyeViewController.init(alertType: .toast,
-                                viewType: .custom(customRyeView),
-                                at: .bottom(inset: 16),
-                                timeAlive: 2)
+let customView = YourCustomViewHere()
+let rye = RyeViewController.init(viewType: .custom(customView),
+                                 at: .bottom(inset: 16))
 rye.show()
 
 ```
 
-### Alert Type
+### Dismiss Completion
+If you would like to execute some code when the Rye alert is dismissed you can pass a `dismissCompletion` code block when calling `show` like so:
+
+```swift
+import Rye
+...
+let ryeConfiguration: RyeConfiguration = [Rye.Configuration.Key.text: "Message for the user"]
+let rye = RyeViewController.init(viewType: .standard(configuration: ryeConfiguration),
+                                 at: .bottom(inset: 16))
+rye.show(withDismissCompletion: {
+    print("Goodbye from Rye, time to dy..die")  
+})
+```
+
+### Dismiss Rye Alerts Manually
+
+If you have selected a to show a Rye alert as `.nonDismissable` you have to dismiss it yourself. Keep a reference to the `RyeViewController` and call `dismiss` when you are ready to let go.
+
+```swift
+import Rye
+...
+var rye: RyeViewController?
+
+let ryeConfiguration: RyeConfiguration = [Rye.Configuration.Key.text: "Message for the user"]
+rye = RyeViewController.init(dismissMode: .nonDismissable,
+                             viewType: .standard(configuration: ryeConfiguration),
+                             at: .bottom(inset: 16))
+rye?.show()
+
+...at a later point in time
+rye?.dismiss()
+```
+
+### Descriptions of Parameters
+
+Below you can find descriptions of the various parameters used to control a Rye alert.
+
+#### Display Modes
+
+Rye supports three different `displayMode` values which can be passed when creating a new `RyeViewController`:
 
-Rye allows you to define the alert type you want to display to your user. Possible alert types are:
+- `automatic`: The alert appears and disappears automatically after a specified interval.
```

Comment:
Now I think about it, I am not sure this is the perfect way to go.

If I choose `.automatic` I'll still want to be able to swipe or tap it away, as per native notification behavior.

So maybe there's only 2 modes - `.default` where if you set dismiss time to e.g. nil or 0 it'll stay until swiped away, and `.nonDismissable` where it'll stay there until dismissed in code. The latter I'd say is very rare even. Even for missing connectivity I believe the better experience is to not be forced to look at a fixed notification.

Would something that took this UX into consideration maybe be desirable?

Reply:
Or maybe just keeping the current config and adding dismiss by gesture to `.automatic` is better.

Reply:
> So maybe there's only 2 modes - .default where if you set dismiss time to e.g. nil or 0 or .largestFiniteMagnitude it'll stay until swiped away

If that is the way we chose to go I prefer to set dismiss time to `nil`. Seems less "magic" or more obvious depending on how you look at it I think. I mean... if we went with 0 being "gesture mode" I could easily hear myself think "OK now I've set the dismiss time to 0...what does that mean? Will it disappear right away then?" 🤔

Reply:
> Or maybe just keeping the current config and adding dismiss by gesture to .automatic is better.

That could also work and I think I like that better.

It would force you to reason about how you want your toast to behave. I get that most of the time you probably go for `.automatic` but in those rare cases where you want the toast to be `.gesture` driven you could do that and it would be pretty obvious in your code.

Reply:
Yeah, makes sense. The 0 thing is just Apple convention, usually 0 is the magic disable value, I think :)

Reply:
There we go:

1a0dfa4518bd246be9bb534ad6b290cdc5520b1a adds a gesture recogniser when in `.automatic` mode as well